authors/apps/articles/views.py

```diff
@@ -270,4 +275,95 @@ def delete(self, request, pk, format=None):
         self.check_object_permissions(request, rating)
         rating.delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
-    
\ No newline at end of file
+    
+    
+class FavoriteView(generics.CreateAPIView):
+    """This class creates a new favorite"""
+    permission_classes = (IsAuthenticated,)
+    serializer_class = FavoriteSerializer
+
+    def post(self, request, slug):
+        """Creates a favorite"""
+        data = request.data
+        article = Articles.objects.filter(
+            # slug=slug, published=True, activated=True).first()
+            slug=slug).first()
+
+        if article is None:
+            not_found = {
+                "detail": "This article has not been found."
+            }
+            return Response(data=not_found, status=status.HTTP_404_NOT_FOUND)
+
+        favorite = Favorite.objects.filter(
+            user_id=request.user.pk, article_id=article.id).first()
+        if favorite is not None:
+            favorite_found = {
+                "detail": "Article already in favorites."
+            }
+            return Response(data=favorite_found,
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        data['article_id'] = article.id
+        data['user_id'] = request.user.pk
+        serializer = self.get_serializer(data=data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        data = serializer.data
+        data['detail'] = 'Article added to favorites.'
+        return Response(data, status=status.HTTP_201_CREATED)
+
+    def delete(self, request, slug):
+        article = Articles.objects.filter(
+            # slug=kwargs['slug'], published=True, activated=True).first()
```

Comment:
Remove this line of commented out code; as well as anywhere else this occurs

Reply:
Noted